{{cookiecutter.project_slug}}/Makefile

```diff
@@ -18,6 +18,11 @@ network:
 build:
 	docker-compose build
 
+## Recompile requirements and store pinned dependencies with hashes.
+pip-compile:
+	docker-compose run --rm web pip-compile --generate-hashes dev-requirements.in
```

Comment:
My suggestion is to use the base image and you have to compile everything at once in order to satisfy all dependencies (as well as notice conflicts).

```makefile
docker run --rm wsgi-base:master pip-compile ...
```

I'm too lazy to find the full command right now but you should mount the local directory and then compile `requirements.in` and `dev-requirements.in` at the same time with corresponding output files.
```

Reply:
> you have to compile everything at once in order to satisfy all dependencies (as well as notice conflicts)

True - then `wsgi-requirements.in` should also be included in the compilation, instead of simply using the precompiled `wsgi-requirements.txt` for `pip-sync`. I actually like that better, since it means that we then update the wsgi dependencies at compile time in the service, instead of having to do that in the base image.

> compile `requirements.in` and `dev-requirements.in` at the same time with corresponding output files

Note that pip-compile generates only a single output file despite multiple inputs, but I think it's fine to simply generate a `requirements.txt`.

Reply:
Although then we will of course lose some of the benefits of the prebuilt image.

Reply:
Hmm, I forgot about those. Problem is that you would need to rebuild the extensions and thus need build tools again... Since wsgi-base runs daily that should be good enough, I think?